elastic/eui
# [EUI+] Split up several component pages into subpages

## Summary

The primary goal of this PR is making certain very long component doc pages more readable/dev-friendly, and also not have a million props tables in a row (EuiButton is the primary culprit of this).

If you have any other suggestions for components that would benefit from more subpages, feel free to throw them into this PR! (or we can open other follow up PRs!)
## QA

Check that the following docs pages and their copy/examples/props table generally work and look sane:

- Button:
  - https://eui.elastic.co/pr_8181/new-docs/docs/navigation/button/
  - https://eui.elastic.co/pr_8181/new-docs/docs/navigation/button/empty/
  - https://eui.elastic.co/pr_8181/new-docs/docs/navigation/button/icon/
  - https://eui.elastic.co/pr_8181/new-docs/docs/navigation/button/group/
- Badge:
  - https://eui.elastic.co/pr_8181/new-docs/docs/display/badge/
  - https://eui.elastic.co/pr_8181/new-docs/docs/display/badge/beta/
  - https://eui.elastic.co/pr_8181/new-docs/docs/display/badge/notification/
- Flex:
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flex/
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flex/group
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flex/item
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flex/grid
- Panel:
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/panel/split/
- Flyout:
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flyout/resizable/
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/flyout/flyout_push/
- Modal:
  - https://eui.elastic.co/pr_8181/new-docs/docs/layout/modal/confirm/
- Steps:
  - https://eui.elastic.co/pr_8181/new-docs/docs/navigation/steps/horizontal/
- Date picker:
  - https://eui.elastic.co/pr_8181/new-docs/docs/forms/date-picker/range/

### General checklist

N/A, docs only
